gdk/commands/component/PublishCommand.py

```diff
@@ -20,35 +20,67 @@ def __init__(self, command_args) -> None:
 
     def run(self):
         try:
-            self.project_config["account_number"] = self.get_account_number()
-            if self.arguments["bucket"]:
-                self.project_config["bucket"] = self.arguments["bucket"]
-            else:
-                self.project_config["bucket"] = "{}-{}-{}".format(
-                    self.project_config["bucket"], self.project_config["region"], self.project_config["account_number"]
-                )
-
+            self.is_project_built()
+            self._update_project_config_values()
             component_name = self.project_config["component_name"]
-            component_version = self.get_component_version_from_config()
-            logging.debug(f"Checking if the component '{component_name}' is built.")
-            if not utils.dir_exists(self.project_config["gg_build_component_artifacts_dir"]):
-                logging.warning(
-                    f"The component '{component_name}' is not built.\nSo, building the component before publishing it."
-                )
-                component.build({})
-            logging.info(f"Publishing the component '{component_name}' with the given project configuration.")
-            logging.info("Uploading the component built artifacts to s3 bucket.")
-            self.upload_artifacts_s3(component_name, component_version)
-
-            logging.info(f"Updating the component recipe {component_name}-{component_version}.")
-            self.update_and_create_recipe_file(component_name, component_version)
-
-            logging.info(f"Creating a new greengrass component {component_name}-{component_version}")
-            self.create_gg_component(component_name, component_version)
+            component_version = self.project_config["component_version"]
+            self._publish_component_version(component_name, component_version)
         except Exception as e:
             logging.error("Failed to publish new version of the component '{}'".format(self.project_config["component_name"]))
             raise Exception("{}\n{}".format(error_messages.PUBLISH_FAILED, e))
 
+    def is_project_built(self):
+        # TODO: This method should just warn and proceed. It should not build the component.
```

Comment:
Should it continue though? Maybe it should stop instead of trying to publish if not suitable version for deployment is found.

Reply:
There are cases when there can be no local artifacts. For eg. 
- All the artifacts in the recipe are already present on s3. (because we first lookup artifact at s3 uri and then locally).
- Component with no artifacts but just a recipe. 

So, in such cases, publish should still continue if there is `greengrass-build/artifacts` directory.